ecs-agent/tmds/handlers/fault/v1/handlers/handlers_test.go

```diff
@@ -347,8 +324,170 @@ func generateNetworkBlackHolePortTestCases(name string, expectedHappyResponseBod
 	return tcs
 }
 
+func generateStartBlackHolePortFaultTestCases() []networkFaultInjectionTestCase {
+	commonTcs := generateNetworkBlackHolePortTestCases("start blackhole port")
+	tcs := []networkFaultInjectionTestCase{
+		{
+			name:                 "start blackhole port success running",
+			expectedStatusCode:   200,
+			requestBody:          happyBlackHolePortReqBody,
+			expectedResponseBody: types.NewNetworkFaultInjectionSuccessResponse("running"),
+			setAgentStateExpectations: func(agentState *mock_state.MockAgentState) {
+				agentState.EXPECT().GetTaskMetadata(endpointId).
+					Return(happyTaskResponse, nil).
+					Times(1)
+			},
+			setExecExpectations: func(exec *mock_execwrapper.MockExec, ctrl *gomock.Controller) {
+			},
+		},
+		{
+			name:               "start blackhole unknown request body",
+			expectedStatusCode: 200,
+			requestBody: map[string]interface{}{
+				"Port":        port,
+				"Protocol":    protocol,
+				"TrafficType": trafficType,
+				"Unknown":     "",
+			},
+			expectedResponseBody: types.NewNetworkFaultInjectionSuccessResponse("running"),
+			setAgentStateExpectations: func(agentState *mock_state.MockAgentState) {
+				agentState.EXPECT().GetTaskMetadata(endpointId).
+					Return(happyTaskResponse, nil).
+					Times(1)
+			},
+			setExecExpectations: func(exec *mock_execwrapper.MockExec, ctrl *gomock.Controller) {
+			},
+		},
+	}
+
+	return append(tcs, commonTcs...)
+}
+
+func generateStopBlackHolePortFaultTestCases() []networkFaultInjectionTestCase {
+	commonTcs := generateNetworkBlackHolePortTestCases("stop blackhole port")
```

Comment:
should we use the same name as the API call? `start network-blackhole-port`

Reply:
hm, don't think I quite follow here. this is for stop black hole port

Reply:
Sorry for the confusion. I was referring to the API name. Should we include the dashes? i.e. `blackhole port` vs `network-blackhole-port`

Reply:
ah gotcha. I was basing it off of the following -> https://github.com/aws/amazon-ecs-agent/blob/dev/ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go#L38-L40